Modules/Devices/src/Devices.Infrastructure/PushNotifications/DirectPush/DirectPushService.cs

```diff
@@ -98,4 +98,28 @@ public async Task UpdateRegistration(IdentityAddress address, DeviceId deviceId,
             }
         }
     }
+
+    public async Task DeleteRegistration(IdentityAddress address, DeviceId deviceId, PnsHandle handle, string appId, CancellationToken cancellationToken)
+    {
+        Console.WriteLine("DirectPushService :: DeleteRegistration");
+
+        var registration = await _pnsRegistrationRepository.FindByDeviceId(deviceId, cancellationToken, track: true);
+
+        if (registration != null)
+        {
+            _logger.LogInformation("Device '{deviceId}' is not found.", deviceId);
+        }
+        else
+        {
+            try
+            {
+                await _pnsRegistrationRepository.Delete(new List<DeviceId> { deviceId }, cancellationToken);
+                _logger.LogTrace("Device '{deviceId}' successfully deleted.", deviceId);
+            }
+            catch (InfrastructureException exception) when (exception.Code == InfrastructureErrors.UniqueKeyViolation().Code)
```

Comment:
Did you simply copy&paste the code from the update method without thinking about what's actually written? Why would there be a "UniqueKeyViolation" in case of a Delete?